### DISCUSSION TITLE
4 eyes principle for configuration changes

### DISCUSSION THREAD

**Discussion by mffap** [zitadel/zitadel#1686] [Ideas]
Should we offer an option to allow configuration changes only after approval by a second [Project / Organisation / IAM Manager](https://docs.zitadel.ch/docs/manuals/admin-managers)?

Potential benefits: 
- Avoid configuration mistakes (eg, deletion of assets, granting of access rights to external persons)
- Less risk of misuse of manager roles (eg, misconfiguration of app url)

**Comment by adlerhurst** [comment]
I understand your points and i think we should have a closer  look at it, maybe after q2 as it's already a lot of work in this milestone 

Would you treat changes by a service users the same way? The potential problem there is that you have an automated process but have to confirm the action manually.

**Reply by fforootd** [reply]
Right now I am not really a "fan" of such a feature. This because of the following points:

- It leaks business logic into the IAM domain (IMO the business process should be part of the IAM)
  -  This somehow would evolve the "platform" into a "workflow engine" and I don't like the idea of this
- The integration with automated process e.g things manipulated by using service accounts looks tricky 

What I think that we should put on the roadmap is instead the feature to `undo changes`

**Reply by adlerhurst** [reply]
What do you think of a notification system like the one of github? You could subscribe for events / objects and have a look at it as a post step without the need of a workflow engine.

**Reply by fforootd** [reply]
I would love to have such subscriptions. we could think of using webhooks for external validations 😁 in the future this would make this whole thing plugable
